### PR TITLE
SP-5822 simple-kubernetes-playbooks

### PR DESCRIPTION
**Pull request by talbe**
### General
- [x] Go over texts with @stackpulse/product
- [x] Explicitly discuss the _playbook name_, _image_ and _badges_
- [x] Explicitly consider various _benefits_. As a rule of thumb, there should be 2-3 of them
- [x] If the playbook has Slack outputs use the Slack experience template

### New playbooks
- [x] Tested Playbook in a production tenant
- [x] Added a link in the [README.MD](https://github.com/stackpulse/playbooks/blob/master/README.md)

### Playbook structure
- [x] For improved human-readability, please add a new empty line between the steps
- [x] Please add a one-liner comment block before each step, for example: `# Retrieve long connections from the database server`
- [x] When considering step IDs, please use short, but descriptive terms, explaining what is being done, such as `retrieve_connections_data` or `update_storage_size`
- [x] When writing params descriptions, start with a capital letter and phrase as a short sentence, such as `User name for database connection` or `Address of the database cluster`

### CODE REVIEW

File: kubernetes/delete-pod/playbook.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  description: Kubernetes rollout restart
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:4]: Wrong description

File: kubernetes/delete-pod/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Kubernetes Delete Pod
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/delete-pod/playbook.yaml)
+
+This playbook performs pod deletion
```
Comment (hagaishapira) [inline kubernetes/delete-pod/README.md:9]: Please keep the how to get started in all READMEs

File: kubernetes/label-logs/playbook.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  description: Kubernetes label logs
```
Comment (hagaishapira) [inline kubernetes/label-logs/playbook.yaml:4]: The descriptions should be a bit more explanatory of what this playbook is doing

**Review by hagaishapira — COMMENTED** [review]
In general we need more extensive descriptions on the playbooks themselves

File: kubernetes/label-logs/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Kubernetes Fetch Logs by Label
```
Comment (hagaishapira) [inline kubernetes/label-logs/README.md:1]: I would change the path to be "logs-by-label" as it's clearer than "label-logs"

File: kubernetes/label-logs/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/label-logs/playbook.yaml)
+
+This playbook fetch logs by label
```
Comment (hagaishapira) [inline kubernetes/label-logs/README.md:9]:
```suggestion
This playbook fetches logs by label
```

**Review by hagaishapira — CHANGES_REQUESTED** [review]
We need all of these to output their content somewhere. 
Please add slack with either the logs, or confirmation for the rollout / pod deletion

File: kubernetes/delete-pod/playbook.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  - name: KubernetesIntegration
+    type: integration
+    description: Kubernetes integration name
+    options:
+      integration_type: k8s
+steps:
+  - name: us-docker.pkg.dev/stackpulse/public/kubectl/delete
+    id: kubectl_cmd
+    env:
+      RESOURCE_TYPE: "pod"
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:22]: What about the more advanced grace parameters?

File: kubernetes/delete-pod/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+## How to get started
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/delete-pod/playbook.yaml)
+
+Not a Stackpulse user? Follow these steps:
+
+1. Make sure your have a [StackPulse](https://stackpulse.com/get-started) account
+2. Import [this playbook](https://app.stackpulse.io/playbooks) into your account
+4. Connect it to monitoring alerts or execute manually
```
Comment (hagaishapira) [inline kubernetes/delete-pod/README.md:19]: What about 3?

File: kubernetes/delete-pod/README.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# Kubernetes Delete Pod
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/delete-pod/playbook.yaml)
+
+This playbook performs pod deletion
```
Comment (hagaishapira) [inline kubernetes/delete-pod/README.md:9]: This text is a bit laconic. Please Change this and the other to be more expressive. 
i.e. "This playbook allows you to delete a forcefully delete a pod without any grace period. This enables you to delete even stuck or hanging pods."

File: kubernetes/rollout-restart/README.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+## How to get started
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/rollout-restart/playbook.yaml)
+
+Not a Stackpulse user? Follow these steps:
+
+1. Make sure your have a [StackPulse](https://stackpulse.com/get-started) account
+2. Import [this playbook](https://app.stackpulse.io/playbooks) into your account
+4. Connect it to monitoring alerts or execute manually
```
Comment (hagaishapira) [inline kubernetes/rollout-restart/README.md:19]: What about 3?

File: kubernetes/logs/README.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+## How to get started
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/logs/playbook.yaml)
+
+Not a Stackpulse user? Follow these steps:
+
+1. Make sure your have a [StackPulse](https://stackpulse.com/get-started) account
+2. Import [this playbook](https://app.stackpulse.io/playbooks) into your account
+4. Connect it to monitoring alerts or execute manually
```
Comment (hagaishapira) [inline kubernetes/logs/README.md:19]: What about 3?

File: kubernetes/logs-by-label/README.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+## How to get started
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/logs-by-label/playbook.yaml)
+
+Not a Stackpulse user? Follow these steps:
+
+1. Make sure your have a [StackPulse](https://stackpulse.com/get-started) account
+2. Import [this playbook](https://app.stackpulse.io/playbooks) into your account
+4. Connect it to monitoring alerts or execute manually
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/README.md:19]: What about 3?

File: kubernetes/rollout-restart/README.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# Kubernetes Rollout Restart
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/rollout-restart/playbook.yaml)
+
+This playbook performs rollout restart for given kubernetes entity
```
Comment (hagaishapira) [inline kubernetes/rollout-restart/README.md:9]:
```suggestion
This playbook performs rollout restart for a given Kubernetes deployment.
```

**Review by hagaishapira — CHANGES_REQUESTED** [review]
What about the Slack steps to send over the outputs?

File: kubernetes/delete-pod/README.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# Kubernetes Delete Pod
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/delete-pod/playbook.yaml)
+
+This playbook allows you to delete forcefully a pod without any grace period. This enables you to delete even stuck or hanging pods.
```
Comment (hagaishapira) [inline kubernetes/delete-pod/README.md:9]:
```suggestion
This playbook allows you to forcefully delete a pod without any grace period. This enables you to delete even stuck or hanging pods.
```

File: kubernetes/delete-pod/playbook.yaml [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
+  - name: force
+    type: var
+    description: whether
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:15]: missing description

File: kubernetes/delete-pod/playbook.yaml [hunk 15]
```diff
@@ -0,0 +1,55 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  description: Kubernetes rollout restart
+  name: kubernetes-delete-pod
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:9]:
```suggestion
    description: The namespace of the pod we wish to delete
    default: default
```

File: kubernetes/delete-pod/playbook.yaml [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:12]: if this is delete pod then what does entity mean? This should be podName or pod with the description saying that this should be a name

File: kubernetes/delete-pod/playbook.yaml [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+      flag to the number of seconds to wait for the Pod to terminate before
+      forcibly terminating it.
+  - name: KubernetesIntegration
+    description: Kubernetes integration name
+    type: integration
+    options:
+      integration_type: k8s
+  - name: slackRecipient
+    type: var
+    description: The recipient for the playbook result notification
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:31]: add a default of "{{ $.metadata.userEmail }}"

File: kubernetes/delete-pod/playbook.yaml [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+      RESOURCES_NAMES: "{{ $.params.entity }}"
+      NAMESPACE: "{{ $.params.namespace }}"
+      GRACE_PERIOD: "{{ $.params.gracePeriod }}"
+      FORCE: "{{ $.params.force }}"
+    envFrom:
+      integrationRef: "{{ $.params.KubernetesIntegration }}"
+  - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+    id: slack_send_snippet
+    env:
+      CONTENT: "The request to delete the entity {{ $.params.entity}} ended with the result {{ $.kubectl_cmd }} "
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:51]:
```suggestion
      CONTENT: "The request to delete the pod {{ $.params.entity}} ended with the result {{ $.kubectl_cmd }} "
```

File: kubernetes/logs-by-label/README.md [hunk 19]
```diff
@@ -0,0 +1,19 @@
+# Kubernetes Fetch Logs by Label
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/logs-by-label/playbook.yaml)
+
+This playbook allows you to fetch logs by given label
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/README.md:9]:
```suggestion
This playbook allows you to fetch logs by a given label.
For example, getting logs from all pods with the label `app=mysql`.
```

File: kubernetes/logs-by-label/playbook.yaml [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+kind: Playbook
+metadata:
+  description: Kubernetes playbook to fetch logs by label
+  name: kubernetes-logs-by-label
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate
+    default: "default"
+    example: "default"
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/playbook.yaml:11]: Do we have example for params?

Comment (hagaishapira) [reply kubernetes/logs-by-label/playbook.yaml:11]: I don't think so

File: kubernetes/logs-by-label/playbook.yaml [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+    required: false
+  - name: label
+    type: var
+    description: The relevant entity to operate on
+    required: false
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/playbook.yaml:16]: I don't think we have required too

Comment (hagaishapira) [reply kubernetes/logs-by-label/playbook.yaml:16]: We do have optional, but if you use that make sure the playbook actually handles that gracefully

File: kubernetes/logs-by-label/playbook.yaml [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate
+    default: "default"
+    example: "default"
+    required: false
+  - name: label
+    type: var
+    description: The relevant entity to operate on
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/playbook.yaml:15]: Fix description for label

File: kubernetes/logs/playbook.yaml [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+kind: Playbook
+metadata:
+  description: Kubernetes playbook to fetch logs from given entity
+  name: kubernetes-logs
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate.
+    default: "default"
+    example: "default"
```
Comment (hagaishapira) [inline kubernetes/logs/playbook.yaml:11]: again go over all of these as we don't have examples and required in parameters

File: kubernetes/logs/playbook.yaml [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+    required: "false"
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
+  - name: allContainers
```
Comment (hagaishapira) [inline kubernetes/logs/playbook.yaml:16]: What do we expect users to input here? what's the default?

File: kubernetes/rollout-restart/playbook.yaml [hunk 25]
```diff
@@ -0,0 +1,44 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  description: Kubernetes playbook to operate rollout restart for given entity
+  name: kubernetes-rollout-restart
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate
```
Comment (hagaishapira) [inline kubernetes/rollout-restart/playbook.yaml:9]:
```suggestion
    description: The relevant namespace to operate in
```

Comment (hagaishapira) [reply kubernetes/rollout-restart/playbook.yaml:9]: either user optional if that works gracefully or have a `default: default`

File: kubernetes/rollout-restart/playbook.yaml [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
```
Comment (hagaishapira) [inline kubernetes/rollout-restart/playbook.yaml:12]:
```suggestion
    description: The relevant entity to operate on, ie. `deployment/nginx`.
```

**Review by hagaishapira — CHANGES_REQUESTED** [review]
See comments and suggestions

File: kubernetes/delete-pod/playbook.yaml (hunk 16 above)
Comment (talbe) [reply kubernetes/delete-pod/playbook.yaml:13]: kubectl delete command can delete not only pods.. you can delete services for example.. actually any kubernetes entity

Comment (talbe) [reply kubernetes/delete-pod/playbook.yaml:13]: NVM 
this is delete pod playbook
will update

File: kubernetes/logs-by-label/playbook.yaml (hunk 20 above)
Comment (talbe) [reply kubernetes/logs-by-label/playbook.yaml:11]: no we don't

File: kubernetes/logs-by-label/playbook.yaml (hunk 21 above)
Comment (talbe) [reply kubernetes/logs-by-label/playbook.yaml:16]: I will set as required

File: kubernetes/logs/playbook.yaml (hunk 24 above)
Comment (talbe) [reply kubernetes/logs/playbook.yaml:16]: boolean value
will add default

File: kubernetes/delete-pod/playbook.yaml [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+  - name: namespace
+    type: var
+    description: The namespace of the pod we wish to delete
+    default: default
+  - name: entity
+    type: var
+    description: The relevant pod to operate on
+  - name: force
+    type: var
+    description: deletes pod without any grace period.
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:16]: What should the user input here? This is free text, we don't have a bool type parameter

File: kubernetes/delete-pod/playbook.yaml [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+kind: Playbook
+metadata:
+  description: Kubernetes rollout restart
+  name: kubernetes-delete-pod
+parameters:
+  - name: namespace
+    type: var
+    description: The namespace of the pod we wish to delete
+    default: default
+  - name: entity
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:11]:
```suggestion
  - name: pod
```

File: kubernetes/logs-by-label/playbook.yaml (hunk 21 above)
Comment (hagaishapira) [reply kubernetes/logs-by-label/playbook.yaml:13]: but we don't have required, by default everything is required. We only have optional to set as "non required"

File: kubernetes/logs/playbook.yaml [hunk 29]
```diff
@@ -0,0 +1,46 @@
...
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate in
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
+  - name: allContainers
+    type: var
+    default: "false"
+    description: Whether fetch logs for all containers (--all-containers flag).
```
Comment (hagaishapira) [inline kubernetes/logs/playbook.yaml:16]: In general when you have free text for parameters that should actually be boolean please specify in the description, "must be either `true` or `false`"

File: kubernetes/rollout-restart/playbook.yaml [hunk 30]
```diff
@@ -0,0 +1,44 @@
+apiVersion: stackpulse.io/v1
+kind: Playbook
+metadata:
+  description: Kubernetes playbook to operate rollout restart for given entity
+  name: kubernetes-rollout-restart
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate in, ie. `deployment/nginx`.
```
Comment (hagaishapira) [inline kubernetes/rollout-restart/playbook.yaml:9]:
```suggestion
    description: The relevant namespace to operate in.
    default: default
```

**Review by hagaishapira — CHANGES_REQUESTED** [review]

File: kubernetes/delete-pod/playbook.yaml [hunk 31]
```diff
@@ -0,0 +1,57 @@
...
+  - name: namespace
+    type: var
+    description: The namespace of the pod we wish to delete
+    default: default
+  - name: pod
+    type: var
+    description: The relevant pod to operate on
+  - name: force
+    type: var
+    description: deletes pod without any grace period. must be either true or false.
```
Comment (hagaishapira) [inline kubernetes/delete-pod/playbook.yaml:16]:
```suggestion
    description: Deletes pod without any grace period. Must be either `true` or `false`.
```

File: kubernetes/logs-by-label/playbook.yaml [hunk 32]
```diff
@@ -0,0 +1,41 @@
...
+metadata:
+  description: Kubernetes playbook to fetch logs by label
+  name: kubernetes-logs-by-label
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate in
+  - name: label
+    type: var
+    description: The relevant label to filter by
```
Comment (hagaishapira) [inline kubernetes/logs-by-label/playbook.yaml:12]:
```suggestion
    description: The relevant label to filter by (for example `app=mysql`).
```

File: kubernetes/logs/playbook.yaml [hunk 33]
```diff
@@ -0,0 +1,46 @@
...
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate in
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
+  - name: allContainers
+    type: var
+    default: "false"
+    description: Whether fetch logs for all containers (--all-containers flag). must be either true or false.
```
Comment (hagaishapira) [inline kubernetes/logs/playbook.yaml:16]:
```suggestion
    description: Whether fetch logs for all containers (--all-containers flag). Must be either `true` or `false`.
```

File: kubernetes/logs/README.md [hunk 34]
```diff
@@ -0,0 +1,19 @@
+# Kubernetes Fetch Logs
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create?tab=playbook#https://github.com/stackpulse/playbooks/blob/master/kubernetes/logs/playbook.yaml)
+
+This playbook allows you to fetch logs from kubernetes entity, with all the options that kubernetes expose to you.
```
Comment (hagaishapira) [inline kubernetes/logs/README.md:9]:
```suggestion
This playbook allows you to fetch logs from a Kubernetes entity, with all the options that Kubernetes expose to you.
```

File: kubernetes/logs/playbook.yaml [hunk 35]
```diff
@@ -0,0 +1,46 @@
...
+metadata:
+  description: Kubernetes playbook to fetch logs from given entity
+  name: kubernetes-logs
+parameters:
+  - name: namespace
+    type: var
+    description: The relevant namespace to operate in
+  - name: entity
+    type: var
+    description: The relevant entity to operate on
```
Comment (hagaishapira) [inline kubernetes/logs/playbook.yaml:12]:
```suggestion
    description: The relevant entity to operate on. If the entity (pod or job) has multiple containers, consider using `allContainers: true` 
```

**Review by hagaishapira — APPROVED** [review]
Merge latest comments

**Pull request merged**